client/src/plugins/system/DataCard.tsx

```diff
@@ -163,19 +177,20 @@ const DataCard: FC<DataCardProps & React.HTMLAttributes<HTMLDivElement>> = (prop
     });
   }
 
-  const systemTitle = [t('systemTitle'), t('valueTitle')];
-  const systemContent = [];
-  const sysInfos = node?.infos?.system_info || {};
+  if (system_configurations) {
+    Object.keys(system_configurations).forEach(key => {
+      systemConfig.push({ label: key, value: system_configurations[key] });
```

Comment:
In this components, why we don't use useEffect to do data transform? It  looks like all data transform operations running every render .

Reply:
This component is a pure component without any user action. So re-render means data changes.